### PR TITLE
Adds Hivemind Link ability, ends Traitorling

### PR DESCRIPTION
**Pull request by Robustin**
1) Traitorling is ruled out during Traitorchan mode. It's too frustrating trying to make lings into a consistent antag when they're under constant assault from snowflake rules and every balance change has to incorporate the absurd discussion of "well what if they're a traitorling...". I'm not wild about this change, but it's necessary to make real progress.

2) Lings can now use holoparasite injectors.

3) Lings can now interrogate their victims with a new default ability, "Hivemind Link". Following a tight (kill) grip, lings can now bring people into the ling hivemind for the purpose of interrogation. Whether you just want to taunt the victim or have them barter valuable intel for their lives, this ability will allow you to 'silently' and somewhat stealthily communicate with conscious or unsconscious people.

Not only is it a neat idea, I created it to compensate for the divide between traitors/lings that this PR will create. While lings won't be starting with traitor gear anymore, they can use this ability to coerce traitors for PDA codes or demand their cooperation in return for sparing their life. The traitorling thing wasn't my first preference (see my old PR) but I'd rather have more freedom balance for Traitor AND Ling without having to address the rare traitorling. Considering we've got as far as cutting traitor items during lowpop, it seems like common sense that we'd take this step.

:cl: Robustin
rscadd: Lings have a new default ability, Hivemind Link. This lets you bring a neck-grabbed victim into hivemind communications after short period of time. You can talk to victims in crit or muted and perhaps persuade them to help you, give you intel, or even a PDA code. The link will stabilize crit victims to ensure they do not die during the link.
tweak: Lings no longer face any unique item restrictions.
rscdel: Lings can no longer be selected as traitors during traitorchan. 
/:cl:

Ability in action: http://imgur.com/5qZMffz

### CODE REVIEW

**Comment by Robustin** [comment]
This is #16497 with requested changes due to repo shenanigans.

**Comment by Niknakflak** [comment]
Robustin attempt number #273748391047472929 to let lings use holoparasite injectors

**Comment by Robustin** [comment]
Attempt #2 actually, and its a compromise because lings can't access them without admin memes, dumb luck, or the use of the new Hivemind Link (tm) Ability!

I'm pro-traitorling, so this wasn't an easy decision for me to reach, but "remove traitorling" has been the consensus in recent threads instead of continuously snowflaking their item selection.

**Comment by PKPenguin321** [comment]
Atomize your features you shitter

File: code/game/gamemodes/changeling/powers/linglink.dm [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		feedback_add_details("changeling_powers","A[stage]")
+		if(!do_mob(user, target, 20))
+			user << "<span class='warning'>Our link with [target] has ended!</span>"
+			changeling.islinking = 0
+			return
+
+	changeling.islinking = 0
+	target.mind.linglink = 0
+	user << "<span class='notice'>You cannot sustain the connection any longer, your victim fades from the hivemind</span>"
+	target << "<span class='userdanger'>The link cannot be sustained any long, your connection to the hivemind has faded!</span>"
```
Comment (Razharas) [inline code/game/gamemodes/changeling/powers/linglink.dm]: Any longER

File: code/modules/mob/living/say.dm [hunk 2]
```diff
@@ -39,21 +39,21 @@ var/list/department_radio_keys = list(
...
+	  ":ò" = "Science",		"#ò" = "Science",		".ò" = "Science",
+	  ":ü" = "Medical",		"#ü" = "Medical",		".ü" = "Medical",
+	  ":ó" = "Engineering",	"#ó" = "Engineering",	".ó" = "Engineering",
+	  ":û" = "Security",	"#û" = "Security",		".û" = "Security",
+	  ":ö" = "whisper",		"#ö" = "whisper",		".ö" = "whisper",
+	  ":è" = "binary",		"#è" = "binary",		".è" = "binary",
+	  ":ô" = "alientalk",	"#ô" = "alientalk",		".ô" = "alientalk",
+	  ":å" = "Syndicate",	"#å" = "Syndicate",		".å" = "Syndicate",
+	  ":é" = "Supply",		"#é" = "Supply",		".é" = "Supply",
+	  ":ï" = "changeling",	"#ï" = "changeling",	".ï" = "changeling"
```
Comment (Razharas) [inline code/modules/mob/living/say.dm]: Dont touch this miracle of encoding plz

File: code/modules/mob/living/say.dm [hunk 3]
```diff
@@ -272,11 +289,13 @@ var/list/crit_allowed_modes = list(MODE_WHISPER,MODE_CHANGELING,MODE_ALIEN)
 			return NOPASS
 	return 0
 
-/mob/living/lingcheck() //Returns 1 if they are a changeling. Returns 2 if they are a changeling that can communicate through the hivemind
+/mob/living/lingcheck() //1 is ling w/ no hivemind. 2 is ling w/hivemind. 3 is ling victim being linked into hivemind.
```
Comment (Razharas) [inline code/modules/mob/living/say.dm]: Only carbons should be able to be lings

**Comment by AdamElTablawy** [comment]
there really isn't a reason for a ling to use holoparas in the first place seeing as if they fuck up  and die that's it GG permanently no more reviving

**Comment by Robustin** [comment]
I agree, but that's why this is a compromise PR.

PKP I put this all in one PR because they're inter-related.

File: code/modules/mob/living/say.dm (hunk 2 above)
Comment (Robustin) [inline code/modules/mob/living/say.dm]: Yea nothing should be changed.

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (Robustin) [inline code/modules/mob/living/say.dm]: But that's irrelevant here right? This is just the communications check, the Linglink.dm and Changeling.dm already ensure non-carbons don't have any way to be Ling'd or join the Hivemind.

**Comment by Shadowlight213** [comment]
@Robustin Incorrect actually. a traitor xenobiologist could become a ling via headslug memes, and then give himself a holo. So it's not admin only.

File: code/modules/mob/living/say.dm (hunk 2 above)
Comment (Core0verload) [inline code/modules/mob/living/say.dm]: Я тебе ебало расхуячу за такое "nothing". Stop breaking encodings!

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (Core0verload) [inline code/modules/mob/living/say.dm]: Non-carbon lings exist, they are called headslugs.

File: code/modules/mob/living/say.dm (hunk 2 above)
Comment (optimumtact) [inline code/modules/mob/living/say.dm]: Christ what the fuck is this

Comment (Core0verload) [inline code/modules/mob/living/say.dm]: Support for russian letters as channel keys.

**Comment by Robustin** [comment]
@Shadowlight213 

I classify that as "dumb luck". Getting rainbow potions AND a headslug is extremely hard.

File: code/modules/mob/living/say.dm (hunk 2 above)
Comment (CPTANT) [inline code/modules/mob/living/say.dm]: The fuck does this have to do with lings?

**Comment by Shadowlight213** [comment]
Tbh, we should just remove headslug. It's caused more issues than its worth.

**Comment by PKPenguin321** [comment]
>I classify that as "dumb luck". Getting rainbow potions AND a headslug is extremely hard

Kind of like rolling traitor and changeling at the same time

**Comment by coiax** [comment]
Except getting a headslug and rainbow slime actually takes time to set up and do, leaving you vulnerable to any murdering/early shuttle call in the meantime.

File: code/modules/mob/living/say.dm (hunk 3 above)
Comment (Razharas) [inline code/modules/mob/living/say.dm]: Never heard of any headsluts

File: code/modules/mob/living/say.dm (hunk 2 above)
Comment (Razharas) [inline code/modules/mob/living/say.dm]: I have no idea why robustin gave this code any attention but good man rastaf0 did great job at making comms available for us for free

**Comment by Robustin** [comment]
No more meme logic please. This PR is almost a month old, I recreated it so we could move forward, not spout the tired old awful tropes. Xenobio lings are extremely rare, I've NEVER seen someone get RAINBOW and GOLDS that get a HEADSLUG. I could maxcap the station for a fraction of the effort, it's a non-issue.

**Comment by PKPenguin321** [comment]
>meme logic
>"its a m-meme guys ignore what he says"

okay well here are my thoughts
1. ling ability is literally useless unless you're gonna be some kind of friendly ling (good luck)
2. lings probably shouldnt be able to use holos since its admittedly too good
3. traitorling is pretty rare and pretty fun, leave it in

**Comment by coiax** [comment]
1. We should probably be encouraging anything that makes people less likely to murderbone, and being able to actually make people bargain for their lives while in crit is cool.
2. Why would a changeling use an item that destroys their body if they die and increase their surface area? It sounds good initially, but holos paint a target on you that changelings don't generally like.
3. It's not fun, it's winning the grief lottery, as suddenly you have double the resources, and the weaknesses of neither. It's dumb luck, but it's not a roll that people have to actively make, it's just something that happens to them.

**Comment by Robustin** [comment]
Coiax nailed it on comment #2 PKP. Lings are stealthy and difficult to permanently kill, Holo's gimp both of these advantages while giving the crew twice as many targets to laser. Not saying they're not strong, but I'm infinitely more afraid of a ling who grabs an ebow and dissonant shriek.

**Comment by Core0verload** [comment]
Make this ability a separate PR. The rest may burn in hell.

**Comment by PKPenguin321** [comment]
1. literally everything you just said is bullshit because why would anybody in their right mind not only let a person know they are a ling, attack them to the point of chokeholding, and then instead of murdering them they GIVE THEM ACCESS TO THEIR EXCLUSIVE PRIVATE CHANNEL AND THEN LET THEM GO? the only scenario this would ever be used in is for the fags who think its fun to gang up with security and out all the other lings, which is, in a nutshell, not fun

2. you are underestimating the strength of having innate and on-demand rapid heals combined with a holoparasite which is balanced around you not having those things.

3. sometimes you gamble and win and it can be fun. the difference between being a strong traitorling and being a traitorling who bought a holoparasite is purely one of game balance like i already said. lings with ebows is really strong, but that's okay because that's not ling-with-holo levels of strong.

idk if i conveyed that too well, kinda tired, but tldr: removing traitorling = removing fun, adding holoparasites to lings = adding anti-fun, adding the dumb """"interrogation"""" (metafriending with sec) ability = adding super anti-fun

:-1: once again

**Comment by Robustin** [comment]
1) Lings have codenames on the channel bro, chill. Lings can decide exactly that they do or don't want to share with the victim.

2) Ling is out'ing himself to the victim but that's usually the case anyway because you know, armblades. It's still their choice and if the Ling doesn't get what he wants from the victim, the victim's never going to wake up to tell anyone. Don't want to reveal your ling nature? DONT USE THE GODDAMN ABILITY THEN PKP SHEESH. 

3) Ling shores up holo's weaknesses but Holo destroys some of ling's strengths. Losing REVIVE is a huge loss for the ling and Ling's ability to stay discrete goes out the window when a holo pops out of you. Plus THEY CANT GET THEM BY DEFAULT WITH TRAITORLING GONE. If a ling gets a holo its because he's earned it. 

The consensus from my oldest PR on this subject was the majority of people just wanted roundstart Traitorlings gone because of the effect it was having on both traitor AND ling balance just for some grief lotto jackpot that happens in a tiny fraction of games.

**Comment by AdamElTablawy** [comment]
At the least it's be useful for

**Comment by AdamElTablawy** [comment]
Yep, you read the above comment correctly.

The ability would be useful for nothing if it goes to the main hivemind rather than a subsection where lings can talk freely to the prisoner at will rather than interrogsting a victim, another ling blabbing he's sucked the captain dry and the victim whl gets spared sacrificing his life to a ling to scream 'CAPTAIN GOT KILLED BY A LING''

**Comment by PKPenguin321** [comment]
robustin
this ability you are adding is shit
more "friendly lings" is not good and if you do that i will revert it

**Comment by Robustin** [comment]
I hate seclings more than any person alive, but if you're dead wrong if you honestly think that:

1) Someone confessing that they're a body stealing changeling who wants to TOTALLY help you  just needs to put you in a fucking chokehold and stab you with a probiscis first is going to persuade almost nobody in security to do anything but drag that asshole to the gibber. Keep in mind that the conditions for linglink are pretty much the same as bodyswap. If someone wants to meme it up with sec I look forward to "friendly" body swapping with the HoS and then muting him. Plus its not like "friendly lings" ever had to get much evidence. Why drag anyone into the hivemind when you can just say "the RD is a ling", hunt him down, and watch as he inevitably does ling shit to try and escape. 

2) Hivemind is going to be some bustling hive of conspiracies, you haven't played many ling rounds. Even if that was the case, the only people lings are likely to spare are traitors or other malcontents who aren't likely to sell the lings out, especially if it means getting a shot at some captain swag/sec gear.

3) To address these concerns I will add some kind of notification when a victim joins the Hivemind.

**Comment by Robustin** [comment]
Just for you PKP: http://imgur.com/5qZMffz

**Comment by PKPenguin321** [comment]
Alright, well now I'll be willing to try it
I suspect it will do more harm than good though

**Comment by Robustin** [comment]
PR Day 35: 

>10 years since robustin PR. i walk through the empty streets trying to think of something else but my path always leads to the PR. i stare at the screen for hours and try to summon the merge. i watch other PR die on page 3 but it is no good. i flame goofball in his PR and try to resist the nazi maintainers but it is all meaningless. the end is near.i then usually watch some old Robustin PRs and cry myself to sleep.

**Comment by KorPhaeron** [comment]
>2) Lings can now use holoparasite injectors.

This is why. Please atomize your PRs to prevent such scenarios from happening.

**Comment by Robustin** [comment]
Becuase they go hand in hand Kor, how many times do I have to explain that. Thats why this is a compromise PR. 

That's why others have FAILED in the past. 

"Lings can now use HP injectors": No
"Lings cannot be roundstart traitors": No
Both: Wow a palatable compromise that gives us a way to move forward on ling and traitor designed without constantly invoking MUH TRAITORLING BALANCE. 

Also, Hivemind link was added EXCLUSIVELY to address concerns about Lings losing their ability to access traitor gear or cooperate with traitors.

**Comment by coiax** [comment]
Gotta love theory crafting balance issues around guardians and changelings. Given how murderboners are common and frequent, it's not like a test merge is going to offer any real difference in the core gameplay experience if it's as bad as it seems.

**Comment by KorPhaeron** [comment]
>Gotta love theory crafting balance issues around guardians and changelings

I think I'm entitlted to theorycraft around something I designed and played extensively.

The stand is unstunnable, so you have to overwhelm it with damage. The only way to deal with it is exhaust it's health pool.

The changeling has an ability that restores its health very very very very quickly. The way to deal with it is to overwhelm it with stuns and exhaust its chems.

So now you have something that can't be stunned with infinite health. Not fun. Not something I want to be yelled at for.

I'll merge the hivelink thing, and maybe the traitorling thing, but definitely not the stand thing.

**Comment by coiax** [comment]
1) If the changeling has managed to get a guardian creator, it deserves it.
**2) Recent nerfs have been applied to Fleshmend, causing diminishing returns, effectively preventing it from gaining more than 10 healing of each damage type a tick.**
3) Damage to a guardian while the summoner is in crit deals clone damage, which no changeling ability is capable of repairing bar the full revive, which cannot be used post death.

**Comment by Robustin** [comment]
I'd reconsider Kor:

https://github.com/tgstation/-tg-station/pull/17306

That got merged when Coiax and I were testing fleshmend this week and realized the "diminishing returns" from last year were almost impossible to trigger. Fleshmend gets much weaker if you start stacking it now, between that and you removing engorged, fleshmend is not the meme it used to be.

The only way a changeling is getting a holoparasite is through incredible luck or incredible persuasion. 

Losing revive is a massive loss for changelings. So many options go out the door with revive gone. As a ling I've been ambushed by traitors (revolvers), explosions, holoparasites, etc. and the go-to tactic is simply play dead, get up, and try again. With a holoparasite your only choice is to sit there and fleshmend any time you get into trouble, and while that isn't a bad tactic, it immediately starts tipping people off that you aren't human. I've taken oddball damage from slimes/plasma/radiation that I didn't have the gene-reset to fix so I simply revive to cure it. 

If we're theory-crafting here then consider that when I'm a traitorling I wouldn't even ~want~ to pursue a holoparasite simply because its a ~35% chance of getting an idiot who's more likely to kill me than help me, and I'd much rather abuse the Dissonant + Ebow combo where nothing short of ballistic weaponry can stop me. Sleeping Carp + Ling is also really strong if you don't have a shot at raiding the armory.

I'll gladly take credit for giving us Ling-Stands because in the <1% of rounds where it occurs, you can just point out that the ling used some incredible means to obtain it. It would be like complaining when a traitor gets dragon blood and becomes an ash drake... it's so rare and so difficult that its really not worth complaining about.

**Comment by coiax** [comment]
Test merge when?

**Comment by ShadowDeath6** [comment]
Doesn't absorb give you their notes?

**Comment by Robustin** [comment]
They get your recent /says, but they do not get memories/notes.

**Comment by Shadowlight213** [comment]
Not getting memories/notes is actually a bug.
I just think no one has gotten around to fixing it.

**Comment by KorPhaeron** [comment]
It might be less awful now with a fleshmend nerf, but I still don't like it or think there is any sort of positive fun gain by risking it. I dont like having to have the same argument over 2 months and 4 PRs either. Please atomize your PRs.

**Pull request closed without merging**